src/tracks/playabletrack/wavetrack/ui/WaveClipTrimHandle.cpp

```diff
@@ -71,13 +77,49 @@ HitTestPreview WaveClipTrimHandle::Preview(const TrackPanelMouseState& mouseStat
 UIHandle::Result WaveClipTrimHandle::Click
 (const TrackPanelMouseEvent& event, AudacityProject* pProject)
 {
+    if (event.event.LeftDown())
+    {
+        mDragStartX = event.event.GetX();
+        mInitialEdgeTime = mTargetBorder == Border::Left ? mClip->GetPlayStartTime()
+            : mClip->GetPlayEndTime();
+    }
     return RefreshCode::RefreshNone;
 }
 
 UIHandle::Result WaveClipTrimHandle::Drag
-(const TrackPanelMouseEvent& event, AudacityProject* pProject)
+(const TrackPanelMouseEvent& event, AudacityProject* project)
 {
-    return RefreshCode::RefreshNone;
+    const auto newX = event.event.GetX();
+    const auto dx = newX - mDragStartX;
+
+    auto& viewInfo = ViewInfo::Get(*project);
+
+    
+    auto t = viewInfo.PositionToTime(viewInfo.TimeToPosition(mInitialEdgeTime, event.rect.x) + dx, event.rect.x);
```

Comment:
This is correct use of PositionToTime and TimeToPosition.   You understatnd that you don't just mutliply dx by some scaling factor.  The assumption of uniform samples per pixel should not be made.  Future projects might change that.